tests/foreman/ui/test_ldap_authentication.py

```diff
@@ -1219,15 +1046,51 @@ def test_external_logout_rhsso(enable_external_auth_rhsso, session):
         update_rhsso_settings_in_satellite(revert=True)
 
 
-@upgrade
 @destructive
-def test_external_new_user_login_and_check_count(session):
+def test_session_expire_rhsso_idle_timeout(enable_external_auth_rhsso, session):
+    """Verify the idle session expiration timeout with external authentication RH-SSO
+
+    :id: 80247b30-a988-11ea-943c-d46d6dd3b5b2
+
+    :steps:
+        1. Change the idle timeout settings for the External Authentication
+        2. Login into Satellite using RHSSO login and wait for the idle timeout
+
+    :expectedresults: After completion of the idle timeout user session
+        should get expired
+    """
+    try:
+        update_rhsso_settings_in_satellite()
+        property_name = 'idle_timeout'
+        setting_entity = entities.Setting().search(query={'search': f'name={property_name}'})[0]
+        setting_entity.value = 1
+        setting_entity.update({'value'})
+
+        with session(login=False):
+            session.rhsso_login.login(
+                {'username': settings.rhsso.rhsso_user, 'password': settings.rhsso.password}
+            )
+            sleep(360)
+            with raises(NavigationTriesExceeded) as error:
+                session.task.read_all(widget_names="current_user")['current_user']
+            assert error.typename == "NavigationTriesExceeded"
+
+    finally:
```

Comment:
Consider using `pytest.request.addfinalizer` to actually define this as cleanup, instead of in a try block within the test.

You can decorate a function defined within the test function at ~L1066 after `setting_entity` is defined.